bids/layout/index.py

```diff
@@ -240,12 +247,13 @@ def _index_file(self, abs_fn, entities):
                 match_vals[e.name] = (e, m)
 
         # Create Entity <=> BIDSFile mappings
+        file_objs = []
         if match_vals:
             for _, (ent, val) in match_vals.items():
                 tag = Tag(bf, ent, str(val), ent._dtype)
-                self.session.add(tag)
+                file_objs.append(tag)
 
-        return bf
+        return bf, file_objs
```

Comment:
Should you not be adding `bf` to `file_objs`, since we were calling `self.session.add(bf)` on L238? Then you can just return `file_objs` here.